src/components/Uploader.js

```diff
@@ -0,0 +1,297 @@
+'use strict';
+
+var d3 = require('d3');
+var store = require('../store');
+var dispatcher = require('../dispatcher');
+var React = require('react');
+
+/**
+ * ## Upload Form Component
+ *
+ * Handles UI and logic of loading data into Data Lasso
+ * either using file or by requesting dataset from a URL
+ */
+var Uploader = React.createClass({
+    getInitialState: function() {
+        return {
+            source: store.get('source'),
+            type: store.get('type'),
+        };
+    },
+
+    componentDidMount: function() {
+        store.on('change:source', () => {this.setState({source: store.get('source')})});
+        store.on('change:type', () => {this.setState({type: store.get('type')})});
+    },
+
+    /**
+     * Handles incoming data either from file uploader form or from
+     * URL form
+     *
+     * @param {string} data - Data as a string, coming from file or URL forms
+     * @param {fn} callback - Callback function for when data was loaded
+     */
+    handleDataLoad: function(data, callback) {
+        if (data) {
+            var parsedData;
+
+            try {
+                parsedData = this.parseFile(data);
+            } catch (e) {
+                callback('Error parsing incoming data');
+            }
+
+            if (parsedData) {
+                store.once('change:entries', () => {
+                    callback();
+                });
+                dispatcher.dispatch({actionType: 'file-uploaded', entries: parsedData});
+            } else {
+                callback('No entries in the data');
+            }
+        } else {
+            callback('Data to load is empty');
+        }
+    },
+
+    /**
+     * Parses incoming data from a string to an object based on the file type provided
+     * @param {string} data - Data as a string, coming from file or URL forms
+     * @returns {object} - Parsed data
+     */
+    parseFile: function (data) {
+        switch (this.state.type) {
+            case 'csv':
+                return d3.csv.parse(data);
+            case 'tsv':
+                return d3.tsv.parse(data);
+            case 'json':
+                return JSON.parse(data);
+        }
+    },
+
+    /**
+     * Renders Uploader UI, with either file upload form, or URL upload form
+     */
+    render: function() {
+        let form = this.state.source === 'file' ? (<FileForm handleDataLoad={this.handleDataLoad}/>) : (<UrlForm handleDataLoad={this.handleDataLoad}/>);
+        return (
+            <div className='uploader'>
+                <Toggles source={this.state.source} type={this.state.type} />
+                {form}
+            </div>
+        )
+    }
+});
+
+/**
+ * ## Toggles React component
+ *
+ * Renders a set of toggles that are used in the Uploader
+ */
+const Toggles = React.createClass({
```

Comment:
That's a good point.

Airbnb React/JSX Style Guide, which seems like a reasonable guide, suggests that all components/classes be in their own files, with exception of stateless functional components.

In light of that (and other style guide instructions), I refactored file structure
 - Each component is in it's file (stateless components excluding)
 - Directories in components correspond to logical separation of concerns in the UI
 - Everything that has to do with 3 dimensional graph was moved to `graph` directory - there was no sense to have that code mixed up with UI code.
 - Got rid of `lib/Class` - a polyfill that is obsolete now that I use es6

Thanks for pointing that out!